packages/network-controller/src/create-auto-managed-network-client.ts

```diff
@@ -209,10 +203,24 @@ export function createAutoManagedNetworkClient<
     networkClient?.destroy();
   };
 
+  const enableRpcFailover = () => {
+    isRpcFailoverEnabled = true;
+    destroy();
+    networkClient = undefined;
```

Comment:
This is the part that I noted as "tricky" in the PR description. AutoManagedNetworkClient is a proxy around NetworkClient, and whenever the provider or block tracker is accessed, a new instance of NetworkClient is created and then cached. To enable or disable the RPC failover functionality, we simply clear any cache that exists, forcing a new instance of NetworkClient to be created on the next access, and we pass along `isRpcFailoverEnabled: true` or `isRpcFailoverEnabled: false` above.